packages/features/insights/server/trpc-router.ts

```diff
@@ -17,6 +18,117 @@ const UserBelongsToTeamInput = z.object({
   isAll: z.boolean().optional(),
 });
 
+type BuildBaseWhereConditionCtxType = {
+  userIsOwnerAdminOfParentTeam: boolean;
+  userOrganizationId: number | null;
+  insightsDb: typeof readonlyPrisma;
+};
+
+interface BuildBaseWhereConditionType {
+  teamId?: number | null;
+  eventTypeId?: number;
+  memberUserId?: number;
+  userId?: number;
+  isAll?: boolean;
+  ctx: BuildBaseWhereConditionCtxType;
+}
+
+const buildBaseWhereCondition = async ({
+  teamId,
+  eventTypeId,
+  memberUserId,
+  userId,
+  isAll,
+  ctx,
+}: BuildBaseWhereConditionType): Promise<{
+  whereCondition: Prisma.BookingTimeStatusWhereInput;
+  isEmptyResponse?: boolean;
+}> => {
+  let whereCondition: Prisma.BookingTimeStatusWhereInput = {};
+  // EventType Filter
+  if (eventTypeId) whereCondition.OR = [{ eventTypeId }, { eventParentId: eventTypeId }];
+  // User/Member filter
+  if (memberUserId) whereCondition.userId = memberUserId;
+  if (userId) {
+    whereCondition.teamId = null;
+    whereCondition.userId = userId;
+  }
+  // organization-wide queries condition
+  if (isAll && ctx.userIsOwnerAdminOfParentTeam && ctx.userOrganizationId) {
+    const teamsFromOrg = await ctx.insightsDb.team.findMany({
+      where: {
+        parentId: ctx.userOrganizationId,
+      },
+      select: {
+        id: true,
+      },
+    });
+    if (teamsFromOrg.length === 0) return { whereCondition, isEmptyResponse: true };
+
+    const teamConditional = {
+      id: {
+        in: [ctx.userOrganizationId, ...teamsFromOrg.map((t) => t.id)],
+      },
+    };
+    const usersFromOrg = await ctx.insightsDb.membership.findMany({
+      where: {
+        team: teamConditional,
+        accepted: true,
+      },
+      select: {
+        userId: true,
+      },
+    });
+    const userIdsFromOrg = usersFromOrg.map((u) => u.userId);
+    whereCondition = {
+      ...whereCondition,
+      OR: [
+        {
+          teamId: {
+            in: [ctx.userOrganizationId, ...teamsFromOrg.map((t) => t.id)],
+          },
+          isTeamBooking: true,
+        },
+        {
+          userId: {
+            in: userIdsFromOrg,
+          },
+          isTeamBooking: false,
+        },
+      ],
+    };
+  }
+
+  if (teamId && !isAll && !eventTypeId) {
+    const usersFromTeam = await ctx.insightsDb.membership.findMany({
+      where: {
+        teamId: teamId,
+        accepted: true,
+      },
+      select: {
+        userId: true,
+      },
+    });
+    const userIdsFromTeam = usersFromTeam.map((u) => u.userId);
+    whereCondition = {
+      ...whereCondition,
+      OR: [
+        {
+          teamId,
+          isTeamBooking: true,
+        },
+        {
+          userId: {
+            in: userIdsFromTeam,
+          },
+          isTeamBooking: false,
+        },
+      ],
+    };
+  }
+  return { whereCondition };
+};
```

Comment:
Extracted the base where clause into a function for scalability and reducing chance of missing out on future filters in different cards/metrics. This also cleans up the code considerably.